new-packages/ts-project/src/state.ts

```diff
@@ -232,8 +266,71 @@ function extractEdgeGroup(
           eventTypeData,
         });
 
+        let seenGuardProp = false;
+
         forEachStaticProperty(ctx, ts, element, (prop, key) => {
           switch (key) {
+            case 'cond':
+              if (seenGuardProp) {
+                // `guard` was already seen
+                ctx.errors.push({
+                  type: 'property_mixed',
+                });
+                return;
+              }
+              if (findProperty(ctx, ts, element, 'guard')) {
+                seenGuardProp = true;
+                return;
+              }
+            // fallthrough
+            case 'guard': {
+              if (seenGuardProp) {
+                ctx.errors.push({
+                  type: 'property_mixed',
+                });
+              }
+              seenGuardProp = true;
+
+              if (ts.isStringLiteralLike(prop.initializer)) {
+                const block = createGuardBlock({
+                  sourceId: prop.initializer.text,
+                  parentId: sourceId,
```

Comment:
why are we setting `node.uniqueId` as `parentId` here? it looks slightly off - how this is important for the Studio? shouldn't a guard be linked to a concrete edge rather than a node?

Reply:
It should be the edge but we didn't have edges at the time of opening this PR :) I need to update it now.

Reply:
Done